### PR TITLE
[#109060102] Use specific containers in concourse tasks

### PR DESCRIPTION
**Pull request by keymon**
[#109060102 Create bootstrap pipeline to deploy BOSH](https://www.pivotaltracker.com/story/show/109060102)

What
----

Using specific containers with the minimum dependencies will help to
understand and track the dependencies added to different tools.

Requirements
------------------

#24 should be merged first to cover a bugfix in s3init.sh

Scope
-----

We use the specific containers `curl-ssl` to do curl and openssl logic and
ruby 2.2.3 instead of the multipurpose toolbox.

We use sh instead of bash in s3init script as the curl-ssl container does
not include bash.

How to test this?
-----------------

Test both pipelines create-deployer and create-microbosh as described in the
README.

It can be pipelines in an already deployed environment, and it is not needed
to wait for the bosh-init tasks to finish, as we only want to test the
tasks modified in this PR.

Who?
---

anyone but @keymon

### CODE REVIEW

File: concourse/scripts/s3init.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
```
Comment (mtekel) [inline concourse/scripts/s3init.sh]: This could break the new, instance-profile enabled s3init, as it is using [[

Comment (keymon) [inline concourse/scripts/s3init.sh]: Then we either:

 * Create a new container including bash.
 * do not use `[[`

I think all of that can be done in the other story about refactor s3init.sh, as it was played later. If we block this PR longer, we block all the other cards.

Comment (dcarley) [inline concourse/scripts/s3init.sh]: The S3 changes have already been merged. Looks like this branch needs a rebase anyway, possibly to pick that up. Could you rebase and test whether it works?

**Comment by keymon** [comment]
@actionjack  #24 should be merged first to cover a bugfix in s3init.sh

Nothing really changes, but we need to rebase from master after #24 is merged.

**Comment by keymon** [comment]
@actionjack #24 has been merged and I rebased this with master. Please do `git pull --rebase` to update your history.

**Pull request merged**